# Tarea Unidad 5 - Actividad 2
Los siguientes datos corresponden a  edad, nivel de colesterol, índice de masa  corporal y presión diastólita de 35 pacientes de un servicio hospitalario:

| Edad | Colesterol | IMC   | Presión |
|------|------------|-------|---------|
| 42   | 282        | 31.64 | 97      |
| 64   | 235        | 30.80 | 90      |
| 47   | 200        | 25.61 | 80      |
| 56   | 200        | 26.17 | 75      |
| 54   | 300        | 31.96 | 100     |
| 48   | 215        | 23.18 | 67      |
| 57   | 216        | 21.19 | 80      |
| 52   | 254        | 26.95 | 70      |
| 67   | 310        | 24.26 | 105     |
| 46   | 237        | 21.87 | 70      |
| 58   | 220        | 25.61 | 70      |
| 62   | 233        | 27.92 | 75      |
| 49   | 240        | 27.73 | 90      |
| 56   | 295        | 22.49 | 95      |
| 63   | 310        | 28.50 | 95      |
| 64   | 268        | 30.04 | 90      |
| 67   | 243        | 23.88 | 85      |
| 49   | 239        | 21.99 | 75      |
| 53   | 198        | 26.93 | 75      |
| 59   | 218        | 27.00 | 85      |
| 65   | 215        | 24.04 | 70      |
| 67   | 254        | 28.65 | 105     |
| 49   | 218        | 25.71 | 85      |
| 53   | 221        | 25.33 | 80      |
| 57   | 237        | 25.42 | 90      |
| 47   | 244        | 23.99 | 85      |
| 58   | 223        | 25.23 | 70      |
| 48   | 198        | 25.81 | 85      |
| 51   | 234        | 26.93 | 80      |
| 49   | 175        | 27.77 | 80      |
| 68   | 230        | 30.85 | 70      |

cont.

| Edad | Colesterol | IMC   | Presión |
|------|------------|-------|---------|
| 58   | 248        | 21.61 | 75      |
| 54   | 218        | 26.3  | 95      |
| 59   | 285        | 31.44 | 100     |
| 45   | 253        | 25    | 75      |

1. Determine el modelo de Regresión Lineal Múltiple.

In [2]:
edad <- c(42, 64, 47, 56, 54, 48, 57, 52, 67, 46, 58, 62, 49, 56, 63, 64, 67, 49, 53, 59, 65, 67, 49, 53, 57, 47, 58, 48, 51, 49, 68, 58, 54, 59, 45)
colesterol <- c(282, 235, 200, 200, 300, 215, 216, 254, 310, 237, 220, 233, 240, 295, 310, 268, 243, 239, 198, 218, 215, 254, 218, 221, 237, 244, 223, 198, 234, 175, 230, 248, 218, 285, 253)
imc <- c(31.64, 30.80, 25.61, 26.17, 31.96, 23.18, 21.19, 26.95, 24.26, 21.87, 25.61, 27.92, 27.73, 22.49, 28.50, 30.04, 23.88, 21.99, 26.93, 27.00, 24.04, 28.65, 25.71, 25.33, 25.42, 23.99, 25.23, 25.81, 26.93, 27.77, 30.85, 21.61, 26.3, 31.44, 25)
presion <- c(97, 90, 80, 75, 100, 67, 80, 70, 105, 70, 70, 75, 90, 95, 95, 90, 85, 75, 75, 85, 70, 105, 85, 80, 90, 85, 70, 85, 80, 80, 70, 75, 95, 100, 75)

pacientes <- data.frame(
    edad,
    colesterol,
    imc,
    presion
)

ml.pacientes <- lm(presion ~ colesterol + imc + edad, data=pacientes)
ml.pacientes$coefficients

(Intercept)  colesterol         imc        edad 
10.74586589  0.18047512  1.04620121  0.03387119

De lo anterior se obtiene la recta:

$ \widehat{y} = 10.7458658882284 + 0.180475119410931x_{1} + 1.0462012095879x_{2} + 0.0338711861623936x_3 $

2.- Determine el coeficiente de determinación ajustado.

In [2]:
sum_ml.pacientes <- summary(ml.pacientes)
coef_det_aj <- sum_ml.pacientes$adj.r.squared
coef_det_aj

[1] 0.4109157

De lo anterior, se obtiene que $R^2 = 0.410915728950569 $ lo cual nos dice que el 41% de la variación puede ser explicada por el modelo.

3.- Determine cuáles de las variables (edad, colesterol e índice de masa corporal) sirven para explicar el valor de la presión diastólica.

In [3]:
sum_ml.pacientes$coefficients

Estimate    Std. Error  t value   Pr(>|t|)    
(Intercept) 10.74586589 16.58335632 0.6479910 0.5217616232
colesterol   0.18047512  0.04624793 3.9023391 0.0004788717
imc          1.04620121  0.50859357 2.0570477 0.0481788646
edad         0.03387119  0.20727501 0.1634118 0.8712548948

In [4]:
n <- nrow(pacientes)
t_tabla <- qt(c(.025, .975), n-4)
t_tabla

[1] -2.039513  2.039513

De lo anterior, obtenemos el valor de tablas de $ t_{[\frac{\alpha}{2}, n-(k+1)]} $ quedando el intervalo:

$(-2.03951344639641, 2.03951344639641)$

Analizando los valores de t se obtiene lo siguiente:

Para la constante:

$ H_{0}: \beta_{0} = 0 \implies \text{La ordenada al origen es 0} $

$ H_{1}: \beta_{0} \neq 0 \implies \text{La ordenada al origen}\textbf{ no}\text{ es 0}$

0.6479910 está _dentro_ del intervalo de t, por tanto <b>no</b> se rechaza la hipótesis nula y concluimos que la ordenada al origen es 0.

Para $x_{1}$:

$ H_{0}: \beta_{1} = 0 \implies \text{El valor de y es independiente del valor de } x_{1} $

$ H_{1}: \beta_{1} \neq 0 \implies \text{El valor de y es dependiente del valor de } x_{1}$

3.9023391 está _fuera_ del intervalo de t, por tanto se rechaza la hipótesis nula y concluimos que la variable $x_{1}$ es buena predictora.

Para $x_{2}$:

$ H_{0}: \beta_{1} = 0 \implies \text{El valor de y es independiente del valor de } x_{2} $

$ H_{1}: \beta_{1} \neq 0 \implies \text{El valor de y es dependiente del valor de } x_{2}$

2.0570477 está _fuera_ del intervalo de t, por tanto se rechaza la hipótesis nula y concluimos que la variable $x_{2}$ es buena predictora.

Para $x_{3}$:

$ H_{0}: \beta_{1} = 0 \implies \text{El valor de y es independiente del valor de } x_{3} $

$ H_{1}: \beta_{1} \neq 0 \implies \text{El valor de y es dependiente del valor de } x_{3}$

0.1634118 está _dentro_ del intervalo de t, por tanto no rechaza la hipótesis nula y concluimos que la variable $x_{3}$ no es buena predictora.

### Probando el nuevo modelo _sin_ la variable _edad_

In [6]:
ml2.pacientes <- lm(presion ~ imc, data=pacientes[,2:4])
ml2.pacientes$coefficients

(Intercept)         imc 
  41.889215    1.574122

De lo anterior se obtiene la recta:

$ \widehat{y} = 11.9434433052844 + 0.182107800746217x_{1} + 1.05725769001108x_{2} $

In [7]:
sum_ml2.pacientes <- summary(ml2.pacientes)
coef_det_aj <- sum_ml2.pacientes$adj.r.squared
coef_det_aj

[1] 0.1558102

De lo anterior, se obtiene que $R^2 = 0.428833032223359 $ lo cual nos dice que $\approx43\%$ de la variación puede ser explicada por el nuevo modelo.